htdocs/user/class/api_users.class.php

```diff
@@ -470,18 +470,13 @@ public function setPassword($id, $send_password = 0, $entity = 1)
 			$entity = (DolibarrApiAccess::$user->entity > 0 ? DolibarrApiAccess::$user->entity : $conf->entity);
 		}
 
-		// I do not like that the first parameter has to be an empty string :-()
-		// in user/card.php it is called like this setPassword($user, '')
-		// and my development server does not actually send email, so I can't really test this
 		$newpassword = $this->useraccount->setPassword($user, '');	// This will generate a new password
 		if (is_int($newpassword) && $newpassword < 0) {
 			throw new RestException(500, 'ErrorFailedToSetNewPassword'.$this->useraccount->error);
 		} else {
 			// Success
 			if ($send_password) {
-				// I do not like that the first parameter has to be an empty string :-(
-				// in user/card.php it is called like this send_password($user, $newpassword)
-				if ($this->useraccount->send_password( $user, $newpassword) > 0) {
+				if ($this->useraccount->send_password($user, $newpassword) > 0) {
```

Comment:
In API context, the $user may be defined as global variable but should be empty.
Can you try to replace with 
DolibarrApiAccess::$user

That is the $user that called the API.

Reply:
@eldy  it was $this->useraccount which was needed and all checks seems to be successful.

I've seen no error with send empty, send false or send true. I can not send though as I have not configured email in my development server.

Also, password does change.